### PR TITLE
Claim remove

### LINKED ISSUE #1783: For Available Articles, dual-role particants should have both 'claim' and 'remove' options
Right now, the remove option is replaced if the user is both instructor and student.

### PR DESCRIPTION
**Pull request by AmitJoki**
Fixes #1783

### CODE REVIEW

**Comment by AmitJoki** [comment]
@ragesoss the build is failing because of `language_switcher.spec` which fails randomly sometimes. Could you please rebuild this?

**Comment by ragesoss** [comment]
This looks like it will show the Select and Remove buttons to any admin, regardless of whether they are also in the student role.

Instead, it should work like this:
* User sees 'Select' if they are a student
* User sees 'Remove' if they are instructor or admin

Those should be independent, so the user would only see both if both conditions apply.

For this, we should be able to use `current_user.isStudent` and `current_user.isNonstudent`, since (unlike user.role, which is just a single value) these not mutually exclusive.

File: app/assets/javascripts/components/articles/available_article.jsx [hunk 1]
```diff
@@ -60,16 +60,21 @@ export const AvailableArticle = createReactClass({
     const ratingMobileClass = `${ratingClass} tablet-only`;
     const articleLink = <a onClick={this.stop} href={article.url} target="_blank" className="inline">{article.formatted_title}</a>;
 
-    let actionButton;
+    let actionCell;
     // Show 'Select' button to students
     if (this.props.current_user.role === 0) {
```
Comment (AmitJoki) [inline app/assets/javascripts/components/articles/available_article.jsx]: @ragesoss this checks for "Students"

File: app/assets/javascripts/components/overview/salesforce_link.jsx [hunk 2]
```diff
@@ -16,6 +16,12 @@ const SalesforceLink = createReactClass({
     ServerActions.linkToSalesforce(this.props.course.id, salesforceId);
   },
 
+  updateSalesforceId() {
```
Comment (ragesoss) [inline app/assets/javascripts/components/overview/salesforce_link.jsx]: Unrelated change.

File: app/assets/javascripts/components/articles/available_article.jsx [hunk 3]
```diff
@@ -60,16 +60,27 @@ export const AvailableArticle = createReactClass({
...
+        <td className="table-action-cell">
+          <button className="button dark" onClick={this.onRemoveHandler}>{I18n.t('assignments.remove')}</button>
+        </td>
+      );
+    } else if (this.props.current_user.isNonStudent && this.props.current_user.isStudent) {
+      actionCell = (
+        <td className="table-action-cell">
+          <button className="button dark" onClick={this.onSelectHandler}>{I18n.t('assignments.select')}</button>
+          <button className="button dark" onClick={this.onRemoveHandler}>{I18n.t('assignments.remove')}</button>
+        </td>
```
Comment (ragesoss) [inline app/assets/javascripts/components/articles/available_article.jsx]: The logic looks right, but this could be done more clearly and with less duplication if the two buttons were set independently of each other. Maybe just leave the `td` there but empty if neither button is present?

File: app/assets/javascripts/components/overview/salesforce_link.jsx (hunk 2 above)
Comment (AmitJoki) [reply app/assets/javascripts/components/overview/salesforce_link.jsx]: ugh! Mixed it up with salesforce id branch. Will remove.

**Comment by AmitJoki** [comment]
Have made the requested changes!

**Comment by ragesoss** [comment]
It's still `if ... else if ... else if`.

I meant that it should just have two independent `if` statements:

`if (isStudent) { set the select button }`
`if (isNonstudent) { set the delete button }`

File: test/components/articles/available_article.spec.jsx [hunk 4]
```diff
@@ -7,7 +7,7 @@ describe('AvailableArticle', () => {
   const props = {
     course: { home_wiki: { language: 'en', project: 'wikipedia' } },
     assignment: { article_title: 'two' },
-    current_user: { role: 0 }, // student role
+    current_user: { isStudent: true }, // student role
```
Comment (AmitJoki) [inline test/components/articles/available_article.spec.jsx]: How do I do such a thing for `course_page_spec.rb:279`?

Comment (ragesoss) [reply test/components/articles/available_article.spec.jsx]: You create two CoursesUsers records, one for each role.

Comment (ragesoss) [reply test/components/articles/available_article.spec.jsx]: There shouldn't be any changes to the ruby specs needed, that I can think of.

Comment (AmitJoki) [reply test/components/articles/available_article.spec.jsx]: `course_page_spec.rb` was failing somehow.

Comment (AmitJoki) [reply test/components/articles/available_article.spec.jsx]: Is my code alright in the first place? With the replacing the whole thing with just two mutually exclusive if conditions?

Comment (ragesoss) [reply test/components/articles/available_article.spec.jsx]: Not sure. I'll check it out today.

Comment (AmitJoki) [reply test/components/articles/available_article.spec.jsx]: I think, I should just use role itself. 

if(role == 0)
  showSelect
if(role == 1 || course.user.admin)
  showRemove

**Comment by AmitJoki** [comment]
@ragesoss it was some problem with `isStudent`,`isNotStudent`. I used the vanilla `course.current_user.role` and it worked out fine.

**Comment by AmitJoki** [comment]
@ragesoss  The build is failing randomly, can you please rebuild this. Yesterday it was passing.

**Comment by AmitJoki** [comment]
Build is successful :+1:

File: app/assets/javascripts/components/articles/available_article.jsx [hunk 5]
```diff
@@ -60,15 +60,16 @@ export const AvailableArticle = createReactClass({
     const ratingMobileClass = `${ratingClass} tablet-only`;
     const articleLink = <a onClick={this.stop} href={article.url} target="_blank" className="inline">{article.formatted_title}</a>;
 
-    let actionButton;
-    // Show 'Select' button to students
+    let actionSelect;
+    let actionRemove;
     if (this.props.current_user.role === 0) {
```
Comment (ragesoss) [inline app/assets/javascripts/components/articles/available_article.jsx]: Using current_user.role is not going to be compatible with showing both at once, except for admins.

If a user is enrolled with both the instructor and student roles in the same course, but is not an admin, then they should see both buttons. In that case, the role will probably be 1, but isStudent and isNonstudent should both be true.

Comment (AmitJoki) [reply app/assets/javascripts/components/articles/available_article.jsx]: @ragesoss I got errors using `isStudent` and `isNonStudent` while in specs. That's why I backtracked on this.
